src/pages/items/index.jsx

```diff
@@ -0,0 +1,148 @@
+import { useEffect, useState } from 'react';
+import NavBar from '../../components/NavBar';
+import { GetItems, GetFavoriteItems } from '../../apis/GetItem';
+import BestCard from './components/BestCard';
+import useDeviceSize from '../../hooks/useDeviceSize';
+import ItemCard from './components/ItemCard';
+import { useNavigate } from 'react-router-dom';
+
+export default function ItemPage() {
+  const [products, setProducts] = useState([]);
+  const [favoriteProduct, setFavoriteProduct] = useState([]);
+  const [sortType, setSortType] = useState('recent');
+  const [currentPage, setCurrentPage] = useState(1);
+  const [totalCount, setTotalCount] = useState(0);
+
+  const navigate = useNavigate();
+
+  const { isMobile, isTablet } = useDeviceSize();
+
+  const getProductsPerPage = () => {
+    if (isMobile) return 4;
+    if (isTablet) return 6;
+    return 10;
+  };
+
+  const ProductsPerPage = getProductsPerPage();
+
+  const totalPages = Math.ceil(totalCount / ProductsPerPage);
+
+  const pageGroupSize = 5;
+
+  const currentGroup = Math.ceil(currentPage / pageGroupSize);
+
+  const startPage = (currentGroup - 1) * pageGroupSize + 1;
+
+  const endPage = Math.min(startPage + pageGroupSize - 1, totalPages);
+
+  useEffect(() => {
+    const fetchFavorite = async () => {
+      const { products, totalCount } = await GetItems(
+        currentPage,
+        ProductsPerPage,
+        sortType
+      );
+      setProducts(products);
+      setTotalCount(totalCount);
+    };
+    fetchFavorite();
+  }, [currentPage, ProductsPerPage, sortType]);
+
+  useEffect(() => {
+    const fetchFavorite = async () => {
+      const { products } = await GetFavoriteItems();
+      setFavoriteProduct(products);
+    };
+    fetchFavorite();
+  }, []);
+
+  const topFavoriteProducts = [...favoriteProduct]
+    .sort((a, b) => b.favoriteCount - a.favoriteCount)
+    .slice(0, isMobile ? 1 : isTablet ? 2 : 4);
```

Comment:
❗️ 수정요청
API 호출시 좋아요순으로 sort해서 가지고 오셔서 프론트에서 한번더 sort하실 필요가 없습니다~
slice 메소드는 원복 배열은 변경시키지 않으니 아래처럼 작서하시면 됩니다.

```suggestion
  const topFavoriteProducts = favoriteProduct.slice(0, isMobile ? 1 : isTablet ? 2 : 4);
```